internal/resource/url.go

```diff
@@ -422,14 +423,53 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 
 	sess.Config.Region = aws.String(region)
 
+	// Determine the bucket and key based on the URL scheme
+	var bucket, key string
+	switch u.Scheme {
+	case "s3":
+		bucket = u.Host
+		key = u.Path
+	case "arn":
+		fullURL := u.Scheme + ":" + u.Opaque
+		if !arn.IsARN(fullURL) {
+			return fmt.Errorf("invalid ARN format of the given URL %s", fullURL)
```

Comment:
Let's just return the canned `ErrInvalidS3ARN` errors rather than creating our own errors here.  Validation should have caught these before we got here anyway.

Reply:
Done